lib/spack/spack/solver/asp.py

```diff
@@ -2399,6 +2530,18 @@ class SpecBuilder(object):
         )
     )
 
+    node_regex = re.compile(r"node\(\d,\"(.*)\"\)")
+
+    @staticmethod
+    def root_node(*, pkg: str) -> str:
```

Comment:
This should be renamed to something other than `root_node` (see other comment on the constant in `concretize.lp`)

Reply:
Done in https://github.com/spack/spack/pull/38447/commits/89e7aa990b09df631ed755aeaf676a11f803ac62